sqlalchemy_aio/engine.py

```diff
@@ -293,12 +303,31 @@ def __init__(self, result_proxy, run_in_thread):
         self._result_proxy = result_proxy
         self._run_in_thread = run_in_thread
 
+    async def __aiter__(self):
```

Comment:
Since Python 3.5.2, `__aiter__` should not return an awaitable. It should return an asynchronous iterator (i.e. an object which also implements `__anext__`). By using yield you're trying to use an async generator, but those are Python 3.6 only.

You can add [njsmith/async_generator](https://github.com/njsmith/async_generator) as a dependency and then return an async generator that is passed `_run_in_thread` and `iterator`.